google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -155,12 +169,12 @@ long getInflightWaitSeconds(StreamWriter streamWriter) {
       return connectionWorker().getInflightWaitSeconds();
     }
 
-    TableSchemaAndTimestamp getUpdatedSchema(StreamWriter streamWriter) {
-      if (getKind() == Kind.CONNECTION_WORKER_POOL) {
+    TableSchema getUpdatedSchema(StreamWriter streamWriter) {
+      if (getKind() == Kind.CONNECTION_WORKER) {
+        return connectionWorker().getUpdatedSchema();
+      } else {
         return connectionWorkerPool().getUpdatedSchema(streamWriter);
```

Comment:
Does this breaks the promise of StreamWriter only saw updates when there is a schema update? I think we should be fine to use nano time since it is monotonic on this machine?https://screenshot.googleplex.com/3Qeo9ouZEnehgMR

Reply:
I don't think nanoTime completely works. it is monotonic, but not strictly increasing - i.e. the current code is broken if the first update has the same nanoTime as the creation time, which is completely possible.